reflex/components/radix/themes/color_mode.py

```diff
@@ -110,69 +89,74 @@ class ColorModeIconButton(IconButton):
     @classmethod
     def create(
         cls,
-        light_component: BaseComponent | None = None,
-        dark_component: BaseComponent | None = None,
+        *children,
         position: LiteralPosition | None = None,
         **props,
     ):
         """Create a icon button component that calls toggle_color_mode on click.
 
         Args:
-            light_component: The component to display when color mode is light.
-            dark_component: The component to display when color mode is dark.
+            *children: The children of the component.
             position: The position of the icon button. Follow document flow if None.
             **props: The props to pass to the component.
 
         Returns:
             The button component.
         """
-        pos_values = get_args(LiteralPosition)
+        if children:
+            console.deprecate(
+                feature_name="passing children to color_mode.button",
+                reason=", use color_mode_cond and toggle_color_mode instead to build a custom color_mode component",
+                deprecation_version="0.5.0",
+                removal_version="0.6.0",
+            )
+
+        position_values = get_args(LiteralPosition)
 
         def find(const, var):
             return Var.create_safe(const).contains(var)
 
-        if isinstance(position, Var):
-            props.setdefault(
-                "position",
-                cond(find(pos_values, position), "fixed", ""),
-            )
-            props.setdefault(
-                "bottom",
-                cond(find(["bottom-left", "bottom-right"], position), "2rem", ""),
-            )
+        def set_var_default(prop, default1, default2=""):
```

Comment:
These static inline methods and lookup table could be moved outside the component to avoid redefining them